src/Middleware/src/Headstart.API/Commands/SupplierSync/SupplierSyncCommand.cs

```diff
@@ -26,11 +27,14 @@ public SupplierSyncCommand(AppSettings settings, IOrderCloudClient oc)
             _oc = oc;
         }
 
-        public async Task<JObject> GetOrderAsync(string ID, DecodedToken decodedToken)
+        public async Task<JObject> GetOrderAsync(string ID, OrderType orderType, DecodedToken decodedToken)
         {
-            var supplierID = ID.Split("-")[1];
             var me = await _oc.Me.GetAsync(accessToken: decodedToken.AccessToken);
-            Require.That(decodedToken.CommerceRole == CommerceRole.Seller || supplierID == me.Supplier.ID, new ErrorCode("Unauthorized", $"You are not authorized view this order", 401));
+            var supplierID = ID.Split("-").Length > 1 ? ID.Split("-")[1] : ID;
```

Comment:
This one's a little funky. Since unsubmitted orders may not have a hyphen, this line would break. We don't actually care what the supplierID is at that time, though (we determine it later in the GenericSupplierCommand) but we don't want this to break on quotes.

Reply:
I think I'm following. So you just added that so it wouldn't break for quotes because this same command gets called on orders in admin where you may display either quote orders or non-quote orders